### ISSUE TITLE
Invalid type for parameter ExternalId, value: None

### ISSUE THREAD

**Issue by abaspinar** [Kenza-AI/sagify#112]
Hi, 

When I specify a role on `sagify cloud train`, it fails if I do not specify the `ExternalId` and got the following error.

> Invalid type for parameter ExternalId, value: None, type: <class 'NoneType'>, valid types: <class 'str'>

In boto3 [sts doc](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts.html#STS.Client.assume_role), it is defined as`ExternalId='string'` so we can not pass None.

Can we assume role [here](https://github.com/Kenza-AI/sagify/blob/master/sagify/sagemaker/sagemaker.py#L28) without `ExternalId` if it is not specified from the cli arguments.

Using Sagify 0.20.4

**Comment by ilazakis** [comment]
Hi @abaspinar 👋, users should be able to provide or omit the external id according to their needs. I’ll flag this as a bug and fix asap.

**Comment by ilazakis** [comment]
Offending commit a366da6426abf873ce267cd339df6267add5660d

**Comment by ilazakis** [comment]
Using an empty string default value for external id [here](https://github.com/Kenza-AI/sagify/blob/3f165db155722ca659983fa4db887d1ceb62a75b/sagify/sagemaker/sagemaker.py#L19) should do the trick.

**Comment by abaspinar** [comment]
Hi @ilazakis,
I just tested and empty string is not working with the following error.

> Invalid length for parameter ExternalId, value: 0, valid range: 2-inf

It seems that it needs at least 2 character. I wonder what is the default value for that in botocore side.

**Comment by ilazakis** [comment]
Couldn't find the _Python_ source, but [Go](https://github.com/aws/aws-sdk-go-v2/blob/0c78da59e2e9a4a6df836b3fadcb09e4ee812dd6/service/sts/api_op_AssumeRole.go#L206) also allows for a minimum of 2 characters for `ExternalID`, but only if it is not *nil*.

```go
if s.ExternalId != nil && len(*s.ExternalId) < 2 {
    invalidParams.Add(aws.NewErrParamMinLen("ExternalId", 2))
}
```

I'm pretty sure the *nil/None* approach has worked for me but haven't used it in a while since I do most training using the EC2 instance's profile (which has a role with *SageMaker* access attached to it) without passing a role explicitly. In fact, I've been using a CD4ML solution we will be open-sourcing soon.

Will run a few tests over the weekend but we can always do the following:
```python
if external_id is not None:
    assumedRoleObject = sts_client.assume_role(
                    RoleArn=aws_role,
                    RoleSessionName="SagifySession",
                    ExternalId=external_id)
else:
    assumedRoleObject = sts_client.assume_role(
                    RoleArn=aws_role,
                    RoleSessionName="SagifySession")
```

But it's still weird that these two (the *else* part of the conditional and using ```ExternalId=None``` in the *if* part) would be interpreted differently 🤷‍♂

**Comment by ilazakis** [comment]
I'll push the above and we can take it from there.

**Comment by ilazakis** [comment]
#115 is now merged and available in 0.20.6 @abaspinar Please let me know if the issue persists and I will reopen. Thanks again for reporting!

**Comment by abaspinar** [comment]
Easy and quick fix. Thanks a lot. Can be changed later with the real default value of the botocore `ExternalId` but I also couldn't really find it when I was digging deeper the python sdk, maybe it is under the hood of somethings I am missing. Anyways, really appreciated for the fix.
